packages/next/src/server/base-server.ts

```diff
@@ -1286,10 +1286,10 @@ export default abstract class Server<ServerOptions extends Options = Options> {
         const invokeQuery = req.headers['x-invoke-query']
 
         if (typeof invokeQuery === 'string') {
-          Object.assign(
-            parsedUrl.query,
-            JSON.parse(decodeURIComponent(invokeQuery))
-          )
+          Object.assign(parsedUrl.query, {
```

Comment:
Mix Object.assign and spread operator is duplication of work. You can just add a third argument:

```
          Object.assign(
            parsedUrl.query,
            JSON.parse(decodeURIComponent(invokeQuery)),
            { __nextLocale: parsedUrl.query.__nextLocale }
          )
```

Reply:
@dartess 
Oh you are right ! Thanks !
You can send a PR to my personal Next.js repository for this fix if you want.

Reply:
I'm sorry but no 😅